src/components/PointCloudViewer.tsx

```diff
@@ -0,0 +1,177 @@
+        intensityOffset = cloud.fields(i)?.offset();
+      }
+    }
+
+    if (
+      xOffset === undefined ||
+      yOffset === undefined ||
+      zOffset === undefined
+    ) {
+      console.warn('Unable to find required fields in point cloud data.');
+      return;
+    }
+
+    const step = cloud.pointStep();
+    let num_points = cloud.dataLength() / cloud.pointStep();
+    num_points = num_points / point_skip_ratio;
+    const points = new Float32Array(num_points * 3);
+    const pointIntensity = new Float32Array(num_points * 3);
+
+    for (let i = 0; i < num_points; i += point_skip_ratio) {
+      const cloud_idx = i * step;
+      const point_idx = i * 3;
+
+      // assume fields are of length 4, and are float datatype
+      points[point_idx + 0] = getFloatFromDataArray(
+        cloud.dataArray()!,
+        cloud_idx + xOffset,
+        cloud_idx + xOffset + 4,
+        !cloud.isBigendian()
+      );
+      points[point_idx + 1] = getFloatFromDataArray(
+        cloud.dataArray()!,
+        cloud_idx + yOffset,
+        cloud_idx + yOffset + 4,
+        !cloud.isBigendian()
+      );
+      points[point_idx + 2] = getFloatFromDataArray(
+        cloud.dataArray()!,
+        cloud_idx + zOffset,
+        cloud_idx + zOffset + 4,
+        !cloud.isBigendian()
+      );
+      if (intensityOffset) {
+        pointIntensity[point_idx + 0] = 0.75;
+        pointIntensity[point_idx + 1] = 0.75;
+        pointIntensity[point_idx + 2] =
+          getFloatFromDataArray(
+            cloud.dataArray()!,
+            cloud_idx + intensityOffset,
+            cloud_idx + intensityOffset + 4,
+            !cloud.isBigendian()
+          ) / max_intensity;
+      }
+    }
+    setPointData(points);
+    setPointIntensityData(pointIntensity);
+  }, [max_intensity, point_skip_ratio, data]);
+
+  const pointsPosAttrib = useMemo(
+    () => (
+      <bufferAttribute
+        attachObject={['attributes', 'position']}
+        // can't use props; need to reconstruct to resize buffer
+        args={[pointData, 3, false]}
+        count={pointData.length / 3}
+        onUpdate={(self) => {
+          self.needsUpdate = true;
+        }}
+      />
+    ),
+    [pointData]
+  );
+
+  const pointsColorAttrib = useMemo(
+    () => (
+      <bufferAttribute
+        attachObject={['attributes', 'color']}
+        // can't use props; need to reconstruct to resize buffer
+        args={[pointIntensityData, 3, false]}
+        count={pointIntensityData.length / 3}
+        onUpdate={(self) => {
+          self.needsUpdate = true;
+        }}
+      />
+    ),
+    [pointIntensityData]
+  );
+
+  const T = useMemo(() => {
+    let rot = new Matrix4().makeRotationAxis(
+      new Vector3(1, 0, 0),
+      -Math.PI / 3
+    );
+    rot = rot.multiply(
+      new Matrix4().makeRotationAxis(new Vector3(0, 0, 1), Math.PI / 2)
+    );
+    return rot.multiplyScalar(point_cloud_scale);
+  }, [point_cloud_scale]);
```

Comment:
I think for now you should merge this, but just to put the idea out there:
If we pull this matrix out of PointCloudViewer and just generate a <points> in with points in world space, we could actually drop the PointCloudViewer into both an image card and the viz view.

Reply:
Yeah, that's why I kept the point cloud processing logic in the viewer. for now it was easier to drop it into the imagery tab, because I didn't feel like doing major updates to viz tab, but we can update that in the future.